lib/ae_mdw/db/state.ex

```diff
@@ -20,74 +22,64 @@ defmodule AeMdw.Db.State do
   @typep cache_name() :: atom()
 
   @opaque t() :: %__MODULE__{
-            txn: Database.transaction() | nil,
+            store: Store.t(),
             stats: stats(),
             cache: %{cache_name() => map()}
           }
 
   @spec new() :: t()
-  def new do
-    %__MODULE__{stats: %{}, cache: %{}}
-  end
+  def new, do: %__MODULE__{store: DbStore.new(), stats: %{}, cache: %{}}
 
   @spec commit(t(), [Mutation.t()]) :: t()
-  def commit(state, mutations) do
-    {:ok, txn} = RocksDb.transaction_new()
-    state2 = %__MODULE__{state | txn: txn}
-
-    new_state =
-      mutations
-      |> List.flatten()
-      |> Enum.reject(&is_nil/1)
-      |> Enum.reduce(state2, &Mutation.execute/2)
-
-    :ok = RocksDb.transaction_commit(txn)
-
-    %__MODULE__{new_state | txn: nil}
+  def commit(%__MODULE__{store: prev_store} = state, mutations) do
+    state3 =
+      TxnDbStore.transaction(fn store ->
```

Comment:
Even though it's mentioned that data belonging to the transaction will be read. Could you document the separation of responsibilities between this read and the `MemStore`.

Reply:
I'd like to explain about the MemStore implementation on the MemStore PR. But for now you're right, the `TxnDbStore` will be needed exactly as it is now, without any changes, since it will simply be the Store used to perform the actual db mutations

Reply:
How would be the plan to use `AeMdw.Db.Store` functions implemented by `TxnDbStore`? Shouldn't the `State` use `TxnDbStore` instead of `DbStore` to continue writing to transactions as before?

Reply:
That's what it does on this line of code we're commenting on:
1. Set the State store to be TxnDbStore
2. Execute mutations using current State (which will then use TxnDbStore)
3. Set store back to DbStore after executing them